### ISSUE TITLE
"error: module has missing stability attribute" is insufficiently explained

### ISSUE THREAD

**Issue by jruderman** [rust-lang/rust#103668]
Given the following code ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=85da69f8dd0b5507585ce5c0318bb195)):

```rust
#![feature(slice_from_ptr_range)]
#![feature(slice_ptr_get)]
#![feature(staged_api)]
#![feature(stmt_expr_attributes)]
#![feature(strict_provenance)]

fn foo() {}
fn bar() {}
```

The current output points to the _entire file_ and doesn't tell me what I did wrong:

```
error: module has missing stability attribute
 --> src/lib.rs:1:1
  |
1 | / #![feature(slice_from_ptr_range)]
2 | | #![feature(slice_ptr_get)]
3 | | #![feature(staged_api)]
4 | | #![feature(stmt_expr_attributes)]
... |
7 | | fn foo() {}
8 | | fn bar() {}
  | |___________^
```

If I understand the feature correctly, the output should be:

```
error: expected at least one pub item in this module to be marked as stable or unstable
note: expected because of `#![feature(staged_api)]` on line 3
```

**Comment by fmease** [comment]
No, with *module has missing stability attribute* the compiler means the current module (in this case the root module, the crate root). E.g. if you add `#![stable(feature = "…", since = "…")] below or above the other inner attributes, the error will go away.

**Comment by jruderman** [comment]
Well, it confused me! I had to binary-search to even figure out that `#![feature(staged_api)]` was the line triggering it.

**Comment by fmease** [comment]
> Well, it confused me!

Can't argue against that. Admittedly, the massive span is a bit useless / misleading, even though it's technically correct.

I remember this used to be the same problem with E0601 (missing `main` function) which used to highlight the entire file. Nowadays, we shrink the span to an empty one in this case and add a label suggesting to add a function.

We could do basically the same for the `staged_api` case.

**Comment by fmease** [comment]
Any uncaught misuses of *internal feature* are considered to be the fault of the user and not a bug in rustc. I'm closing this as works as intended / not planned as per [MCP 620](https://github.com/rust-lang/compiler-team/issues/620).
